src/GraphQLinq.Client/GraphQuery.cs

```diff
@@ -182,20 +181,14 @@ public GraphItemQuery<TResult> Select<TResult>(Expression<Func<T, TResult>> resu
             return (GraphItemQuery<TResult>)BuildSelect(resultSelector);
         }
 
-        public abstract T ToItem();
+        public abstract Task<T> ToItem();
     }
 
-    public abstract class GraphCollectionQuery<T> : GraphQuery<T>, IEnumerable<T>
+    public abstract class GraphCollectionQuery<T> : GraphQuery<T>
     {
         protected GraphCollectionQuery(GraphContext graphContext, string queryName) : base(graphContext, queryName) { }
 
-        public abstract IEnumerator<T> GetEnumerator();
-
-        [ExcludeFromCodeCoverage]
-        IEnumerator IEnumerable.GetEnumerator()
-        {
-            return GetEnumerator();
-        }
+        public abstract Task<IEnumerable<T>> ToArray();
```

Comment:
I think it will be weird for users if `ToArray` returns an `IEnumerable`

Reply:
There is no ToArray in `GraphCollectionQuery`.  It implements IEnumerable so that's why you see that method.

Reply:
I don't think having a clear difference in the names here is crucial. In the end, most of the time, we use the 'var' keyword and the IntelliSense to figure out the type. So, I prefer to make names identical, but it is up to you to decide.

Reply:
What about implementing `IAsyncEnumerable` instead of adding a method to `GraphCollectionQuery` ?

Reply:
For what reason? Please clarify why it is necessary. I can not see why we need it.

Reply:
In that case, the `GraphCollectionQuery` does not need to have `Execute` method and can be enumerated directly with async.

Reply:
What benefits does this solution have over just calling a method ```Execute```?

Reply:
The benefit would be that you would be able to enumerate the query itself just like to can do with EF Core query.